mesonbuild/compilers/mixins/clike.py

```diff
@@ -625,9 +625,17 @@ def alignment(self, typename: str, prefix: str, env: 'Environment', *,
             raise mesonlib.EnvironmentException('Could not compile alignment test.')
         if res.returncode != 0:
             raise mesonlib.EnvironmentException('Could not run alignment test binary.')
-        align = int(res.stdout)
-        if align == 0:
-            raise mesonlib.EnvironmentException(f'Could not determine alignment of {typename}. Sorry. You might want to file a bug.')
+
+        align: int
+        try:
+            align = int(res.stdout)
+            if align == 0:
+                raise mesonlib.EnvironmentException(f'Could not determine alignment of {typename}. Sorry. You might want to file a bug.')
+        except ValueError:
+            # If we get here, the user is most likely using a script that is
+            # pretending to be a compiler.
+            raise mesonlib.EnvironmentException('Could not run alignment test binary.')
```

Comment:
Move the except up to after the int(), and in the event that it isn't excepted and raised, we can check `if align == 0`.

It's not meaningfully different except that we avoid catching ValueErrors that theoretically bubble up from the wrong place.